pkg/controllers/node/termination/controller.go

```diff
@@ -77,13 +79,24 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 	if !controllerutil.ContainsFinalizer(node, v1beta1.TerminationFinalizer) {
 		return reconcile.Result{}, nil
 	}
+
+	nodeGracePeriodExpirationTime, err := c.terminationGracePeriodExpirationTime(ctx, node)
```

Comment:
Do we drive the addition of this taint through the Node termination controller or through the NodeClaim termination controller? Since this value is a property of the NodeClaim, I think it would be nice if we drove the addition of the taint from the NodeClaim termination controller when the NodeClaim has been around past its terminationGracePeriod